Python/_warnings.c

```diff
@@ -443,6 +491,174 @@ warn_explicit(PyObject *category, PyObject *message,
     return result;  /* Py_None or NULL. */
 }
 
+static PyObject *
+warn_explicit_with_fix(PyObject *category, PyObject *message,
+              PyObject *fix, PyObject *filename, int lineno,
+              PyObject *module, PyObject *registry, PyObject *sourceline)
+{
+    PyObject *key = NULL, *fix_txt = NULL, *text = NULL, *result = NULL, *lineno_obj = NULL;
+    PyObject *item = Py_None;
+    const char *action;
+    int rc, rc_msg, rc_fix;
+
+    if (registry && !PyDict_Check(registry) && (registry != Py_None)) {
+        PyErr_SetString(PyExc_TypeError, "'registry' must be a dict or None");
+        return NULL;
+    }
+
+    /* Normalize module. */
+    if (module == NULL) {
+        module = normalize_module(filename);
+        if (module == NULL)
+            return NULL;
+    }
+    else
+        Py_INCREF(module);
+
+    /* Normalize message and fix. */
+    Py_INCREF(message);  /* DECREF'ed in cleanup. */
```

Comment:
Does Python provide a way to check that reference counts are zero on termination? Some way we can check that we've got this bit right?

Reply:
I verified that even Python 2.7 had the -R 3:3 option when running tests to check for refleaks, this:

`./python -m test -R 3:3 test_sys`

Uncovers reference counting bugs.